### PR TITLE
Manifest cleanup in preparation for FileProvider changes

### PR DESCRIPTION
**Pull request by lognaturel**
I am looking at designing the API to support external applications that currently directly read from or write to `/sdcard/odk` and won't be able to after #3360. There are currently several Android Studio checks failing and those make it hard to see clearly in the manifest. I'd like to make these quick cleanup changes before starting work.

#### What has been done to verify that this works as intended?
There should be no behavioral changes. I ran all tests and also made sure that the app actually launches.

#### Why is this the best possible solution? Were any other approaches considered?
I split each change into its own commit to explicitly explain why they're being made.

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?
There should be no behavior change. The riskiest part is probably 8573897 but the documentation is very explicit about not needing this block and devices that don't support OpenGL 2 must be quite rare by now.

#### Do we need any specific form for testing your changes? If so, please attach one.
No.

#### Does this change require updates to documentation? If so, please file an issue [here]( https://github.com/opendatakit/docs/issues/new) and include the link below.
I have filed https://github.com/opendatakit/docs/issues/1168 to document backup behavior since it seems like something that users would like to explicitly know about.

#### Before submitting this PR, please make sure you have:
- [x] run `./gradlew checkAll` and confirmed all checks still pass OR confirm CircleCI build passes and run `./gradlew connectedDebugAndroidTest` locally.
- [x] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/opendatakit/collect/blob/master/collect_app/src/main/assets/open_source_licenses.html).
- [x] verified that any new UI elements use theme colors. [UI Components Style guidelines](https://github.com/opendatakit/collect/blob/master/CONTRIBUTING.md#ui-components-style-guidelines)

### CODE REVIEW

File: collect_app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -87,7 +78,8 @@ the specific language governing permissions and limitations under the License.
         android:label="@string/app_name"
         android:largeHeap="true"
         android:supportsRtl="true"
-        android:usesCleartextTraffic="true">
+        android:usesCleartextTraffic="true"
+        android:allowBackup="true">
```
Comment (seadowg) [inline collect_app/src/main/AndroidManifest.xml:82]: I was initially confused why this was added but then I remembered it's true by default. I'm guessing the intention is just to make it explicit here?

**Review by seadowg — APPROVED** [review]
Looks good. Just one question tiny question.

File: collect_app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (lognaturel) [reply collect_app/src/main/AndroidManifest.xml:82]: Yes, exactly. There's an Android Studio check for it and I hadn't actually fully understood the ramifications so I think it's helpful to see it explicitly listed.

**Comment by seadowg** [comment]
I feel like we can merge this without QA. Will do when the build is green again!

**Pull request merged**